pkgs/dartpad_ui/lib/main.dart

```diff
@@ -563,22 +565,31 @@ class LoadingOverlay extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final theme = Theme.of(context);
-    return ValueListenableBuilder<CompilingState>(
-      valueListenable: appModel.compilingState,
-      builder: (_, compilingState, __) {
-        final color = theme.colorScheme.surface;
-        final compiling = compilingState == CompilingState.restarting;
-
-        // If reloading, show a progress spinner. If restarting, also display a
-        // semi-opaque overlay.
-        return AnimatedContainer(
-          color: color.withValues(alpha: compiling ? 0.8 : 0),
-          duration: animationDelay,
-          curve: animationCurve,
-          child:
-              compiling
-                  ? const GoldenRatioCenter(child: CircularProgressIndicator())
-                  : const SizedBox(width: 1),
+    return ValueListenableBuilder<GenAiState>(
+      valueListenable: appModel.genAiManager.state,
+      builder: (BuildContext context, GenAiState genAiState, Widget? child) {
+        return ValueListenableBuilder<CompilingState>(
+          valueListenable: appModel.compilingState,
+          builder: (_, compilingState, __) {
+            final color = theme.colorScheme.surface;
+            final loading =
+                compilingState == CompilingState.restarting ||
+                genAiState == GenAiState.generating;
+
+            // If reloading, show a progress spinner. If restarting, also display a
+            // semi-opaque overlay.
```

Comment:
nit: stay within 80 columns

```suggestion
            // If reloading, show a progress spinner. If restarting, also
            // display a semi-opaque overlay.
```